src/common/copilot/dataverseMetadata.ts

```diff
@@ -76,35 +78,42 @@ export function getEntityName(telemetry: ITelemetry, sessionID:string, dataverse
 
-            const matchingFiles = getMatchingFiles(activeFileFolderPath, fileNameFirstPart); // ["Copilot-Student-Loan-Registration-56a4.basicform.yml"]
+            const matchingFiles = await getMatchingFiles(activeFileFolderPath, fileNameFirstPart); // ["Copilot-Student-Loan-Registration-56a4.basicform.yml"]
 
-            if (matchingFiles[0]) {
+            if (IS_DESKTOP && matchingFiles[0]) {
+                const diskRead = await import('fs');
                 const yamlFilePath = path.join(activeFileFolderPath, matchingFiles[0]);
-                const yamlContent = fs.readFileSync(yamlFilePath, 'utf8');
+                const yamlContent = diskRead.readFileSync(yamlFilePath, 'utf8');
                 const parsedData = parseYamlContent(yamlContent, telemetry, sessionID, dataverseEntity);
                 entityName = parsedData['adx_entityname'] || parsedData['adx_targetentitylogicalname'];
+            } else if (!IS_DESKTOP) {
+                entityName = getFileLogicalEntityName(document.uri.fsPath);
             }
         }
     } catch (error) {
-        sendTelemetryEvent(telemetry, { eventName: CopilotGetEntityFailureEvent, copilotSessionId:sessionID, dataverseEntity: dataverseEntity, error: error as Error});
+        sendTelemetryEvent(telemetry, { eventName: CopilotGetEntityFailureEvent, copilotSessionId: sessionID, dataverseEntity: dataverseEntity, error: error as Error });
         entityName = '';
     }
-
     return entityName;
 }
 
-function getMatchingFiles(folderPath: string, fileNameFirstPart: string): string[] {
-    const files = fs.readdirSync(folderPath);
-    const pattern = new RegExp(`^${fileNameFirstPart}\\.(basicform|list|advancedformstep)\\.yml$`);
-    return files.filter((fileName) => pattern.test(fileName));
+async function getMatchingFiles(folderPath: string, fileNameFirstPart: string): Promise<string[]> {
+    if (IS_DESKTOP) {
+        const diskRead = await import('fs');
+        const files = diskRead.readdirSync(folderPath);
+        const pattern = new RegExp(`^${fileNameFirstPart}\\.(basicform|list|advancedformstep)\\.yml$`);
+        return files.filter((fileName) => pattern.test(fileName));
+    }
+
+    return [];
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-function parseYamlContent(content: string, telemetry: ITelemetry, sessionID:string, dataverseEntity: string): any {
+function parseYamlContent(content: string, telemetry: ITelemetry, sessionID: string, dataverseEntity: string): any {
     try {
         return yaml.parse(content);
     } catch (error) {
-        sendTelemetryEvent(telemetry, { eventName: CopilotYamlParsingFailureEvent, copilotSessionId:sessionID, dataverseEntity: dataverseEntity, error: error as Error });
+        sendTelemetryEvent(telemetry, { eventName: CopilotYamlParsingFailureEvent, copilotSessionId: sessionID, dataverseEntity: dataverseEntity, error: error as Error });
```

Comment:
same for all places where 'dataverseEntity' identifier is used. if it's the name of the entity, than append `Name` to the end of the identifiers.

Like is already done elsewhere with the `logicalEntityName` identifiers.